src/main/java/bio/terra/app/usermetrics/BardClient.java

```diff
@@ -56,4 +78,69 @@ public void logEvent(AuthenticatedUserRequest userReq, BardEvent event) {
       logger.warn("Error logging event {}", event.getEvent(), e);
     }
   }
+
+  /**
+   * Syncing a user is only needed when a new token is not already cached or the cache entry has
+   * expired.
+   *
+   * @param userReq - the AuthenticatedUserRequest that represents the current user.
+   */
+  boolean syncUser(AuthenticatedUserRequest userReq) {
```

Comment:
Is the boolean just used for testing purposes?  If so we probably want a comment to that effect.  Also, there might be other ways to test the interaction if you end up making a change to how the map gets updated below

Reply:
Currently the value is only being used by tests.  This is more out of habit because I try to reduce the amount of spy/mock needed to use to test.  I'm happy to adjust this but for now have added a comment.  Let me know your preference for this and moving going forward.

Reply:
Refactored away when switching to computeIfAbsent.  The mocks give me the insight I need to test this function indirectly.